optimade/validator/validator.py

```diff
@@ -164,27 +164,30 @@ def validate_implementation(self):
 
         # Test entire implementation
         print(f"Testing entire implementation at {self.base_url}...")
-        self._log.debug("Testing base info endpoint of %s", "info")
+        self._log.debug("Testing base info endpoint of %s", CONF.info_endpoint)
 
-        # Save base info to construct new queries
+        # Get and validate base info to find endpoints
         info_endp = CONF.info_endpoint
         base_info = self.test_info_or_links_endpoints(info_endp)
         if not base_info:
-            raise RuntimeError("Unable to understand base info, cannot continue.")
-
-        self.base_info = base_info.dict()
-        self.provider_prefix = self.base_info["meta"].get("provider", {}).get("prefix")
+            raise RuntimeError(
+                "Unable to deserialize base info (see previous errors), so cannot continue."
+            )
 
-        # Run some tests on the versions endpoint
-        versions_endp = CONF.versions_endpoint
-        self.test_versions_endpoint(versions_endp)
-        self.test_bad_version_returns_553()
+        self.provider_prefix = (
+            base_info.dict()["meta"].get("provider", {}).get("prefix")
+        )
 
+        # Set the response class for all `/info/entry` endpoints
         self.available_json_endpoints = self.get_available_endpoints(base_info)
-        # set the response class for all `/info/entry` endpoints
         for endp in self.available_json_endpoints:
             self.response_classes[f"{info_endp}/{endp}"] = EntryInfoResponse
 
+        # Run some tests on the versions endpoint
+        self._log.debug("Testing versions endpoint %s", CONF.versions_endpoint)
+        self.test_versions_endpoint(CONF.versions_endpoint)
```

Comment:
What did you think about removing this method's input argument, using `CONF.versions.endpoint` directly in the `test_versions_endpoint` method?

Reply:
Ambivalent :upside_down_face: